e2e/src/aws_kms.rs

```diff
@@ -0,0 +1,149 @@
+    }
+}
+
+impl AwsKms {
+    pub fn with_show_logs(mut self, show_logs: bool) -> Self {
+        self.show_logs = show_logs;
+        self
+    }
+
+    pub async fn start(self) -> anyhow::Result<AwsKmsProcess> {
+        let container = AwsKmsImage
+            .start()
+            .await
+            .with_context(|| "Failed to start KMS container")?;
+
+        if self.show_logs {
+            spawn_log_printer(&container);
+        }
+
+        let port = container.get_host_port_ipv4(4566).await?;
+        let url = format!("http://localhost:{}", port);
+
+        let credentials = Credentials::new("test", "test", None, None, "Static Test Credentials");
+        let region = Region::new("us-east-1");
+
+        let config = AwsConfig::for_testing(credentials, region, url.clone()).await;
+        let client = AwsClient::new(config);
+
+        Ok(AwsKmsProcess {
+            _container: container,
+            client,
+            url,
+        })
+    }
+}
+
+fn spawn_log_printer(container: &testcontainers::ContainerAsync<AwsKmsImage>) {
+    let stderr = container.stderr(true);
+    let stdout = container.stdout(true);
+    tokio::spawn(async move {
+        let mut stderr_lines = stderr.lines();
+        let mut stdout_lines = stdout.lines();
+
+        let mut other_stream_closed = false;
+        loop {
+            tokio::select! {
+                stderr_result = stderr_lines.next_line() => {
+                    match stderr_result {
+                        Ok(Some(line)) => eprintln!("KMS (stderr): {}", line),
+                        Ok(None) if other_stream_closed => break,
+                        Ok(None) => other_stream_closed = true,
+                        Err(e) => {
+                            eprintln!("KMS: Error reading from stderr: {:?}", e);
+                            break;
+                        }
+                    }
+                }
+                stdout_result = stdout_lines.next_line() => {
+                    match stdout_result {
+                        Ok(Some(line)) => eprintln!("KMS (stdout): {}", line),
+                        Ok(None) if other_stream_closed => break,
+                        Ok(None) => other_stream_closed = true,
+                        Err(e) => {
+                            eprintln!("KMS: Error reading from stdout: {:?}", e);
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok::<(), std::io::Error>(())
+    });
+}
+
+pub struct AwsKmsProcess {
+    _container: testcontainers::ContainerAsync<AwsKmsImage>,
+    client: AwsClient,
+    url: String,
+}
+
+impl AwsKmsProcess {
+    pub async fn create_key(&self) -> anyhow::Result<KmsTestKey> {
+        let response = self
+            .client
+            .inner()
+            .create_key()
+            .key_usage(aws_sdk_kms::types::KeyUsageType::SignVerify)
+            .key_spec(aws_sdk_kms::types::KeySpec::EccSecgP256K1)
```

Comment:
These are fine to come directly from `aws_sdk_kms` instead of `fuels` since they're not part of the `fuels` public API.